scheduler/rank.go

```diff
@@ -206,10 +206,35 @@ OUTER:
 			continue
 		}
 
-		// Index the existing network usage
+		// Index the existing network usage.
+		// This should never collide, sice it represents the current state of
+		// the node. If it does collide though, it means we found a bug! So
+		// collect as much information as possible and send it back to the
+		// server.
```

Comment:
`send it back to the server` is a bit awkward as this is the server! Maybe just "collect as much info as possible." Let the logging/reporting behavior be defined by sendEvent.